### ISSUE TITLE
Getting a warning for 'compile' is obsolete 

### ISSUE THREAD

**Issue by powysm** [BranchMetrics/cordova-ionic-phonegap-branch-deep-linking-attribution#636]
using release 3.4.0 in studio 3.3.2

I get the following message on build

> WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
> It will be removed at the end of 2018.

This is the offending line:

`compile (name:'Branch', ext:'aar')`

Changing to the following resolves:

`implementation (name:'Branch', ext:'aar')`

**Comment by echo-branch** [comment]
That's a maintenance release for users who cannot upgrade.  I believe the 4.x series has that change.  

@codinronan

**Comment by codinronan** [comment]
@powysm @echo-branch That is correct; the 3.x releases are intended to be backwards compatible with older Android Studio + Older Xcode, meaning we aren't changing things like `compile vs implementation` and CocoaPods support.

The `rules.gradle` file was removed in the 4.x releases and uses the defualt Cordova config (which uses `implementation`) so it no longer has that error.

The 4.x releases are 100% up to date and don't have those warnings.
